client/ratings_reviews/components/ReviewFormModal.jsx

```diff
@@ -29,28 +29,46 @@ class ReviewFormModal extends React.Component {
   handleChange(e) {
     let name = e.target.name;
     let value = e.target.value;
-    let photoIncrement = 0;
     if (name.slice(0, 15) === 'characteristics') {
       let currentCharacteristics = this.state.characteristics;
       currentCharacteristics[e.target.className.toString()] = Number.parseInt(value);
       value = currentCharacteristics;
       name = name.slice(0, 15);
     }
-    if (name === 'photo' && value !== '') {
-      let currentPhotos = this.state.photos;
-      currentPhotos.push(value);
-      value = currentPhotos;
-      name = 'photos';
-      photoIncrement++;
-    }
-    let showUploadPhotosButton = this.state.photoCount + photoIncrement < 5;
     this.setState({
-      [name]: value,
-      photoCount: this.state.photoCount + photoIncrement,
-      showUploadPhotosButton: showUploadPhotosButton
+      [name]: value
     });
   }
 
+  handlePhotoUpload(e) {
+    let photo = document.getElementById('review-uploaded-photo').files[0];
+    let reader = new FileReader();
+    let dataURI;
+    reader.onload = () => {
+      dataURI = reader.result;
+      $.ajax({
+        url: '/review/image',
+        method: 'POST',
+        dataType: 'text',
+        data: {
+          dataURI: dataURI
+        }
+      }).then((cloudinaryURL) => {
```

Comment:
This is interesting! Looks great, just want to see if I'm understanding what's happening generally:

1. Create a FileReader instance (have to google this)
2. One it loads, post a URI generated by the FileReader (is this a random one every time?) to the review/image endpoint
3. AJAX call returns a cloudinary URL which is where the new image is saved
4. This gets added to the array of photos in the component state

Reply:
Hi @jabrake, yes! That's my understanding of the process here as well, although I believe that a URI generated by the FileReader should be the same for the same image. Another way to do this might be to send a raw image file to the server and handle multi-part data on the server-side. I may try to refactor to that approach later on if they're much more efficient, but right now this way seems to work too.

Reply:
Psyched about this convo because it would've taken me forever to figure this out. Just a quesiton...does that url '/review/image' work? I think I'm missing something--I always put http://localhostblahblahblah there, too.

Reply:
Hi @matthewwrobel, I'm sure it wouldn't take you long to figure this out! And great question, I don't usually include the localhost part just in case I ever have to change the port number on the server-side (so that I don't have to worry about the AJAX calls on the client side) Either way seems to work, but feel free to let me know if you want me to include the localhost part for consistency.